determined-ai/determined
# feat: Update the "Continue Single trial experiment" workflow

## Description

Implements the "Continue Trial" workflow as described in WEB-1857





## Test Plan
1. Navigate to an experiment list page.
2. Navigate to a completed single trial experiment. 
3. Click "Continue Trial" and ensure you see a dropdown with the options "Create New Experiment..." and "Reactivate Current Trial..."
4. Click "Create New Experiment..." ensure the design and text match the Figma. 
5. Click "Reactivate Current Trial..." ensure the design and text match the Figma. 
6. Click "Create New Experiment..." ensure that the behavior matches the current "Continue Trial" flow (same modal as in step 1)
7. Click "Reactivate Current Trial..." .
8. Click "Show Full Config"  and take note of the current "max_length"
9. Click "Show Simple Config" and input a positive number, check the full config again and make sure that the number was added to the "max_length"
10. Edit the "max_length" in the full config and click show simple config, ensure that the new additional value is equal to the (new  max length - original config max length)
11.  Edit the "max_length" in the full config to be an invalid value, for example a string, or float, or delete the max value entirely and click show simple config, ensure that the additional value field is blank. 
12. In the simple config enter a valid value for "additional batches" and click "Reactivate Trial" ensure that the submissions is successful.
13. Navigate to an incomplete single trial experiment. 
14. Click Continue Trial then "Reactivate Current Trial..." Ensure that the design matches the figma and you can only see the "full config" option and cannot switch to the simple config.
15. Click "Reactivate Trial" ensure that the submissions is successful.
16. Navigate to a completed single trial experiment. Ensure your screen size is large enough to see the "Continue Trial" button. 
17. Click the overflow menu (the three vertical dots) ensure that you do **not** see "Create New Experiment" or "Reactivate Experiment". 
18. Decrease your screen size until you cannot see the "Continue Trial Button" ensure that you can see "Create New Experiment" and "Reactivate Experiment"
19. Ensure you can only see the Continue Trial option for canceled, errored, completed, and/or deleted failed experiments.




## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
WEB-1857
